crates/diesel_models/src/query/role.rs

```diff
@@ -170,4 +199,96 @@ impl Role {
+                };
+                query = query
+                    .filter(dsl::org_id.eq(org_id))
+                    .filter(
+                        dsl::scope
+                            .eq(RoleScope::Organization)
+                            .or(dsl::scope.eq(RoleScope::Merchant))
+                            .or(dsl::scope.eq(RoleScope::Profile)),
+                    )
+                    .filter(dsl::entity_type.eq_any(entity_in_vec))
+            }
+
+            ListRolesByEntityPayload::Merchant(org_id, merchant_id) => {
+                let entity_in_vec = if is_lineage_data_required {
+                    vec![EntityType::Merchant, EntityType::Profile]
+                } else {
+                    vec![EntityType::Merchant]
+                };
+                query = query
+                    .filter(dsl::org_id.eq(org_id))
+                    .filter(
+                        dsl::scope
+                            .eq(RoleScope::Organization)
+                            .or(dsl::scope
+                                .eq(RoleScope::Merchant)
+                                .and(dsl::merchant_id.eq(merchant_id.clone())))
+                            .or(dsl::scope
+                                .eq(RoleScope::Profile)
+                                .and(dsl::merchant_id.eq(merchant_id))),
+                    )
+                    .filter(dsl::entity_type.eq_any(entity_in_vec))
+            }
+
+            ListRolesByEntityPayload::Profile(org_id, merchant_id, profile_id) => {
+                let entity_in_vec = vec![EntityType::Profile];
+                query = query
+                    .filter(dsl::org_id.eq(org_id))
+                    .filter(
+                        dsl::scope
+                            .eq(RoleScope::Organization)
+                            .or(dsl::scope
+                                .eq(RoleScope::Merchant)
+                                .and(dsl::merchant_id.eq(merchant_id.clone())))
+                            .or(dsl::scope
+                                .eq(RoleScope::Profile)
+                                .and(dsl::merchant_id.eq(merchant_id))
+                                .and(dsl::profile_id.eq(profile_id))),
+                    )
+                    .filter(dsl::entity_type.eq_any(entity_in_vec))
+            }
+        };
+
+        if let Some(limit) = limit {
+            query = query.limit(limit.into());
+        }
+
+        router_env::logger::debug!(query = %debug_query::<Pg,_>(&query).to_string());
+
+        match generics::db_metrics::track_database_call::<Self, _, _>(
+            query.get_results_async(conn),
+            generics::db_metrics::DatabaseOperation::Filter,
+        )
+        .await
+        {
+            Ok(value) => Ok(value),
+            Err(err) => match err {
+                DieselError::NotFound => {
+                    Err(report!(err)).change_context(errors::DatabaseError::NotFound)
+                }
```

Comment:
Will this throw error if nothing is found?

Reply:
This case wont come as it will resolve in Ok when value is [] . Hence will remove this